src/Microsoft.AspNetCore.Razor/Chunks/Generators/AddTagHelperChunkGenerator.cs

```diff
@@ -35,5 +37,23 @@ public override void GenerateChunk(Span target, ChunkGeneratorContext context)
         {
             context.ChunkTreeBuilder.AddAddTagHelperChunk(_lookupText, target);
         }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            var other = obj as AddTagHelperChunkGenerator;
+            return base.Equals(obj) &&
+                string.Equals(_lookupText, other._lookupText, StringComparison.Ordinal);
```

Comment:
In the event `obj` is a `SpanChunkGenerator`  but not a `AddTagHelperChunkGenerator` this might be null. Change the first line to `base.Equals(other)` and you should be good.